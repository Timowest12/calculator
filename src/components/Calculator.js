import React, { Component } from "react";

export class Calculator extends Component {
  render() {
    return (
      <div className="calculatorcontainer">
        <div className="outputbar">
          <span className="outputinp">0</span>
        </div>

        <div className="keys">
          <div className="row row1">
            <div className="key key1">ac</div>
            <div className="key key2">+/-</div>
            <div className="key key3">%</div>
            <div className="key key4 orange">/</div>
          </div>
          <div className="row row2">
            <div className="key key1">7</div>
            <div className="key key2">8</div>
            <div className="key key3">9</div>
            <div className="key key4 orange">*</div>
          </div>
          <div className="row row3">
            <div className="key key1">4</div>
            <div className="key key2">5</div>
            <div className="key key3">6</div>
            <div className="key key4 orange">-</div>
          </div>
          <div className="row row4">
            <div className="key key1">1</div>
            <div className="key key2">2</div>
            <div className="key key3">3</div>
            <div className="key key4 orange">+</div>
          </div>
          <div className="row row5">
            <div className="key key1 flex2">0</div>
            <div className="key key2 flex1">1</div>
            <div className="key key3 flex1 orange">=</div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
