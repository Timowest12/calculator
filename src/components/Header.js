import React from 'react'
import {Link} from 'react-router-dom'

function Header() {
    return (
        <header class='nav'>
            <h1>Math Magicans</h1>
            <ul className="navitems">
           
            <Link to="/"><li className="navitem">home</li></Link>
            <Link to="/caculator"><li className="navitem">calculator</li></Link>
            <Link to="/quote"><li className="navitem">quote</li></Link>
            </ul>
        </header>
    )
}

export default Header
