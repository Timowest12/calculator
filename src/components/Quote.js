import React from 'react'

const quote = () => {
    return (
        <div class='screen'>
            <h1>“mathematics is not about numbers, equations, computations or algorithms; it’s about understanding.”</h1>
        </div>
    )
}

export default quote
