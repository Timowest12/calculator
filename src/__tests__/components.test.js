import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../components/Calculator';
import Header from '../components/Header';
import Quote from '../components/Quote';


describe('test screenshots', () => {
  test('calculator screenshot', () => {
    const calculator = TestRenderer.create(<Calculator />).toJSON();

    expect(calculator).toMatchSnapshot();
  });
  test('quote screenshot', () => {
    const quote = TestRenderer.create(<Router>
        <Quote />
      </Router>,).toJSON();

    expect(quote).toMatchSnapshot();
  });
  test('header screenshot', () => {
    const header = TestRenderer.create(
        <Router>
        <Header />
      </Router>,
      ).toJSON();

    expect(header).toMatchSnapshot();
  });
});