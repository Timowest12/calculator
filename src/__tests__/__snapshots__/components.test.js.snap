// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots calculator screenshot 1`] = `
<div
  className="calculatorcontainer"
>
  <div
    className="outputbar"
  >
    <span
      className="outputinp"
    >
      0
    </span>
  </div>
  <div
    className="keys"
  >
    <div
      className="row row1"
    >
      <div
        className="key key1"
        onClick={[Function]}
      >
        ac
      </div>
      <div
        className="key key2"
        onClick={[Function]}
      >
        +/-
      </div>
      <div
        className="key key4"
        onClick={[Function]}
      >
        %
      </div>
      <div
        className="key key3 orange"
        onClick={[Function]}
      >
        ÷
      </div>
    </div>
    <div
      className="row row2"
    >
      <div
        className="key key1"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="key key2"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="key key3"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="key key4 orange"
        onClick={[Function]}
      >
        x
      </div>
    </div>
    <div
      className="row row3"
    >
      <div
        className="key key1"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="key key2"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="key key3"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="key key4 orange"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row row4"
    >
      <div
        className="key key1"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="key key2"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="key key3"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="key key4 orange"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row row5"
    >
      <div
        className="key key1 flex2"
        onClick={[Function]}
      >
        0
      </div>
      <div
        className="key key2 flex1"
        onClick={[Function]}
      >
        .
      </div>
      <div
        className="key key3 flex1 orange"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`test screenshots header screenshot 1`] = `
<header
  className="nav"
>
  <h1>
    Math Magicans
  </h1>
  <ul
    className="navitems"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <li
        className="navitem"
      >
        home
      </li>
    </a>
    <a
      href="/caculator"
      onClick={[Function]}
    >
      <li
        className="navitem"
      >
        calculator
      </li>
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      <li
        className="navitem"
      >
        quote
      </li>
    </a>
  </ul>
</header>
`;

exports[`test screenshots quote screenshot 1`] = `
<div
  className="screen"
>
  <h1>
    “mathematics is not about numbers, equations, computations or algorithms; it’s about understanding.”
  </h1>
</div>
`;
